homeassistant/components/vesync/light.py

```diff
@@ -68,17 +79,88 @@ def turn_on(self, **kwargs):
             brightness = round((brightness / 255) * 100)
             # clamp to 1-100
             brightness = max(1, min(brightness, 100))
-            self.dimmer.set_brightness(brightness)
+            self.device.set_brightness(brightness)
+        # set white temperature
+        if ATTR_COLOR_TEMP in kwargs:
+            # get white temperature from HA data
+            color_temp = int(kwargs[ATTR_COLOR_TEMP])
+            # convert Mireds to Percent value that api expects
+            color_temp = round(
+                ((color_temp - self.min_mireds) / (self.max_mireds - self.min_mireds))
+                * 100
+            )
+            # flip cold/warm to what pyvesync api expects
+            color_temp = 100 - color_temp
+            # ensure value between 0-100
+            color_temp = max(0, min(color_temp, 100))
+            # pass value to pyvesync library api
+            self.device.set_color_temp(color_temp)
         # Avoid turning device back on if this is just a brightness adjustment
         if not self.is_on:
```

Comment:
In this case is_on means it was on the last time we updated the device

If the device is on, and you turn it off outside of HA, then call the turn on service before HA has a chance to update, the light won't turn back on because HA thinks it's already on and then won't send the api call.

Reply:
changed it